feature-find-meal/src/main/java/com/kinandcarta/feature/find/meal/adapter/MealsAdapter.kt

```diff
@@ -1,81 +1,75 @@
-        private val availableView = itemView.findViewById<TextView>(R.id.mealAvailable)
-        private val expiryView = itemView.findViewById<TextView>(R.id.mealExpiry)
-        private val portionsView = itemView.findViewById<TextView>(R.id.mealPortions)
-        private val reserveButton = itemView.findViewById<Button>(R.id.reserveButton)
+    inner class MealsViewHolder(private val itemBinding: MealItemRowBinding): RecyclerView.ViewHolder(itemBinding.root) {
+        private val ctx = itemBinding.root.context
 
         fun bindData(meal: Meal, position: Int) {
-            nameView.text = meal.name
-            val tempString = if (meal.hot) "Hot" else "Cold"
-            val tempColor = if (meal.hot) R.color.colorHot else R.color.colorCold
-            tempView.text = tempString
-            tempView.setTextColor(ContextCompat.getColor(itemView.context, tempColor))
+            itemBinding.mealName.text = meal.name
+            setupMealTemp(meal.hot)
+            setupMealInfo(meal.info)
+            itemBinding.mealDistance.text = "${meal.distance} ${distanceUnit}"
+            itemBinding.mealAvailable.text = "${ctx.getString(R.string.available_leader)} ${meal.availableFromDate}"
+            itemBinding.mealExpiry.text = "${ctx.getString(R.string.expiry_leader)} ${meal.expiryDate}"
+            itemBinding.mealPortions.text = "${meal.quantity.getPortionsString()}"
+            setupReserveButton(meal.quantity, meal.id)
+        }
 
-            if (meal.info.isNullOrEmpty()) {
-                infoView.isVisible = false
+        private fun setupMealTemp(isHot: Boolean) {
+            val tempString = if (isHot) ctx.getString(R.string.hot) else ctx.getString(R.string.cold)
+            val tempColor = if (isHot) R.color.colorHot else R.color.colorCold
+            itemBinding.mealTemp.text = tempString
+            itemBinding.mealTemp.setTextColor(ContextCompat.getColor(itemBinding.root.context, tempColor))
+        }
+
+        private fun setupMealInfo(info: String?) {
+            if (info.isNullOrEmpty()) {
+                itemBinding.mealInfo.isVisible = false
             } else {
-                infoView.text = "Info: ${meal.info}"
+                itemBinding.mealInfo.text = "${ctx.getString(R.string.info_leader)} ${info}"
             }
+        }
 
-            distanceView.text = "${meal.distance} ${distanceUnit}"
 
-            availableView.text = "Available: ${meal.availableFromDate}"
-            expiryView.text = "Expires: ${meal.expiryDate}"
+        private fun setupReserveButton(quantity: Int, mealId: String) {
+            val hasPortions = quantity > 0
+            val reserveButton = itemBinding.reserveButton
 
-            portionsView.text = "#    ${meal.quantity.getPortionsString()}"
+            reserveButton.text = if (hasPortions) ctx.getString(R.string.reserve_button_available) else ctx.getString(R.string.reserve_button_unavailable)
 
```

Comment:
These calls to getString read very long, is there a preferred way to handle this?  Or, in this case, would it be more readable to do:
```
if (hasPortions) {
    reserveButton.text = foo
} else {
   reserveButton.text = bar
}
```

Reply:
You could create another local variable before
```kotlin
val reserveText = if (hasPortions) ctx.getString(R.string.reserve_button_available) else ctx.getString(R.string.reserve_button_unavailable)

...

reserveButton.text = reserverText
```